command/connect/expose/expose.go

```diff
@@ -0,0 +1,238 @@
+	conf, _, err := client.ConfigEntries().Get(api.IngressGateway, gateway, nil)
+	if err != nil && !strings.Contains(err.Error(), agent.ConfigEntryNotFoundErr) {
+		c.UI.Error(fmt.Sprintf("Error fetching existing ingress gateway configuration: %s", err))
+		return 1
+	}
+	if conf == nil {
+		conf = &api.IngressGatewayConfigEntry{
+			Kind:      api.IngressGateway,
+			Name:      gateway,
+			Namespace: gatewayNamespace,
+		}
+	}
+
+	// Make sure the flags don't conflict with existing config.
+	ingressConf, ok := conf.(*api.IngressGatewayConfigEntry)
+	if !ok {
+		// This should never happen
+		c.UI.Error(fmt.Sprintf("Config entry is an invalid type: %T", conf))
+		return 1
+	}
+
+	listenerIdx := -1
+	for i, listener := range ingressConf.Listeners {
+		// Make sure the service isn't already exposed in this gateway
+		for _, service := range listener.Services {
+			if service.Name == svc {
+				c.UI.Error(fmt.Sprintf("Service %q already exposed through listener with port %d", svc, listener.Port))
+				goto CREATE_INTENTION
+			}
+		}
+
+		// If there's already a listener for the given port, make sure the protocol matches.
+		if listener.Port == c.port {
+			listenerIdx = i
+			if listener.Protocol != c.protocol {
+				c.UI.Error(fmt.Sprintf("Listener on port %d already configured with conflicting protocol %q", listener.Port, listener.Protocol))
+				return 1
+			}
+		}
+	}
+
+	// Add a service to the existing listener for the port if one exists, or make a new listener.
+	if listenerIdx >= 0 {
+		ingressConf.Listeners[listenerIdx].Services = append(ingressConf.Listeners[listenerIdx].Services, api.IngressService{
+			Name:      svc,
+			Namespace: svcNamespace,
+		})
+	} else {
+		ingressConf.Listeners = append(ingressConf.Listeners, api.IngressListener{
+			Port:     c.port,
+			Protocol: c.protocol,
+			Services: []api.IngressService{
+				{
+					Name:      svc,
+					Namespace: svcNamespace,
+				},
+			},
+		})
+	}
+
+	// Write the updated config entry using a check-and-set, so it fails if the entry
+	// has been changed since we looked it up.
+	{
+		succeeded, _, err := client.ConfigEntries().CAS(ingressConf, ingressConf.GetModifyIndex(), nil)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error writing ingress config entry: %v", err))
+			return 1
+		}
+		if !succeeded {
+			c.UI.Error("Ingress config entry was changed while attempting to update, please try again.")
+			return 1
+		}
+		c.UI.Output(fmt.Sprintf("Successfully updated config entry for ingress service %q", gateway))
+	}
+
+CREATE_INTENTION:
+	// Check for an existing intention.
+	ixnFinder := finder.Finder{Client: client}
+	existing, err := ixnFinder.Find(c.ingressGateway, c.service)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error looking up existing intention: %s", err))
+		return 1
+	}
+	if existing != nil && existing.Action == api.IntentionActionAllow {
+		c.UI.Error(fmt.Sprintf("Intention already exists for %q -> %q", c.ingressGateway, c.service))
```

Comment:
Similar question as above, should this be `c.UI.Output` instead? Or do you explicitly want it to do to `stderr`?